compiler/zrc_codegen/src/program.rs

```diff
@@ -323,170 +323,64 @@ fn cg_program_without_optimization<'ctx>(
-                    //     name.value(),
-                    //     u32::try_from(n)
-                    //         .expect("should not be more than u32::MAX args in a function"),
-                    //     compilation_unit.get_file(),
-                    //     ident_line_col.line,
-                    //     dbg_ty,
-                    //     true,
-                    //     0,
-                    // );
-
-                    // FIXME: Re-enable this when Inkwell resolves TheDan64/inkwell#613
-                    // dbg_builder.insert_declare_at_end(
-                    //     alloc,
-                    //     Some(decl),
-                    //     None,
-                    //     id_debug_location,
-                    //     entry,
-                    // );
-
-                    fn_scope.insert(name.value(), alloc);
-                }
-
-                cg_block(
-                    FunctionCtx::from_unit_and_fn(unit, fn_value),
-                    entry,
-                    &fn_scope,
-                    lexical_block,
-                    body,
-                    &None,
-                );
-            }
-            // We do not attach debugging information to extern functions, this is clang's behavior
-            // so I assume it's correct.
-            TypedDeclaration::FunctionDeclaration {
-                name,
-                parameters,
-                return_type,
-                body: None,
-            } => {
-                let fn_value = cg_init_extern_fn(
-                    &unit,
-                    name.value(),
-                    return_type.value(),
-                    parameters
-                        .value()
-                        .as_arguments()
-                        .iter()
-                        .map(|ArgumentDeclaration { ty, .. }| ty.value())
-                        .collect::<Vec<_>>()
-                        .as_slice(),
-                    parameters.value().is_variadic(),
-                );
+                // Collect argument types for the function signature
+                let arg_types: Vec<_> = parameters
+                    .value()
+                    .as_arguments()
+                    .iter()
+                    .map(|ArgumentDeclaration { ty, .. }| ty.value())
+                    .collect();
+
+                let is_variadic = parameters.value().is_variadic();
+
+                // Calculate the function value (using if as an expression)
+                let fn_value = if body.is_some() {
+                    // Extract just the value (.0) from the tuple returned by cg_init_fn
+                    cg_init_fn(
+                        &unit,
+                        name.value(),
+                        line_lookup.lookup_from_index(span.start()).line,
+                        return_type.value(),
+                        &arg_types,
+                        is_variadic,
+                    )
+                    .0
```

Comment:
Is there a reason behind only needing the fn_value in contrast to the DISubprogram as needed prior?